### DISCUSSION TITLE
help!  synchronization  of multi-core simulation

### DISCUSSION THREAD

**Discussion by Eric-Zhang-C** [gem5/gem5#1002] [Q&A] [answered]
In gem5's multi-core simulation, each core runs on a separate host thread. In this scenario, how is it ensured that the clocks of these cores are synchronized, avoiding one core's clock from advancing significantly beyond the others? Which classes contain the relevant code?

**Comment by qqli579** [comment, marked as answer]
As far as I know gem5 does not support host multi-threads.There is only one host thread during multi-core simulation.

**Reply by giactra** [reply]
Agree. For the sake of completeness, there have been several attemps to parallelize gem5 but those are not currently part of upstream

**Reply by Eric-Zhang-C** [reply]
thanks for your answer
